### PR TITLE
[CPDP-1526] - Capture contact name of Capacitor and Personal Represen…

### PR DESCRIPTION
**Pull request by modivilioglu**
…tatives

### CODE REVIEW

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala [hunk 1]
```diff
@@ -127,6 +130,85 @@ class RepresenteeControllerSpec
...
   "RepresenteeController" when {
+    "handling change contact name" must {
+      "display the page" when {
+        def performAction(): Future[Result] = controller.changeContactName()(FakeRequest())
+
+        def test(
+          sessionData: SessionData,
+          expectedTitleKey: String,
+          expectedBackLink: Call,
+          expectedPrepopulatedValue: ContactName
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala:141]: it doesn't look like you're using this value

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala [hunk 2]
```diff
@@ -127,6 +130,85 @@ class RepresenteeControllerSpec
...
+      behave like contactNameFormValidationTests(
+        performAction,
+        () =>
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+          }
+      )
+
+      "redirect to check your answers page" in {
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala:189]: Done.

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala (hunk 1 above)
Comment (modivilioglu) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala:141]: Done.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala [hunk 3]
```diff
@@ -458,6 +459,55 @@ class RepresenteeController @Inject() (
...
+                answers.fold(
+                  i => i.copy(contactDetails = i.contactDetails.map(_.copy(contactName = contactName))),
+                  c =>
+                    IncompleteRepresenteeAnswers(
+                      Some(c.name),
+                      Some(c.id),
+                      c.dateOfDeath,
+                      Some(c.contactDetails.copy(contactName = contactName)),
+                      true,
+                      true
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala:498]: Done.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala [hunk 4]
```diff
@@ -458,6 +459,55 @@ class RepresenteeController @Inject() (
         updateDraftReturnAndSession(newAnswers, journey).map(_ => defaultContactDetails)
     }
 
+  def changeContactName(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    val backLink = routes.RepresenteeController.checkContactDetails()
+    withCapacitorOrPersonalRepresentativeAnswers(request) { (_, journey, answers) =>
+      getContactDetails(journey, answers).fold(
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala:465]: Done.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala [hunk 5]
```diff
@@ -458,6 +459,55 @@ class RepresenteeController @Inject() (
         updateDraftReturnAndSession(newAnswers, journey).map(_ => defaultContactDetails)
     }
 
+  def changeContactName(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    val backLink = routes.RepresenteeController.checkContactDetails()
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala:463]: Done.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/change_contact_name.scala.html [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    @caption(messages("representee.caption"))
+
+    @pageHeading(title)
+
+    @formWrapper(routes.RepresenteeController.changeContactNameSubmit(), 'novalidate -> "novalidate") {
+        @textInput(
+            fieldId = "contactName",
+            value = form.data.getOrElse(contactNameKey, ""),
+            errorKey = form.error(contactNameKey).map(e => e.message),
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/change_contact_name.scala.html:75]: It was working but for consistency I fixed them to be same.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/change_contact_name.scala.html [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            label = Html(""),
+            labelClass = Some("govuk-label--m"),
+            helpText = Some(Html(messages(s"$key.change.help"))),
+            autoComplete = AutoCompleteType.ContactName
+
+        )
+
+        @submitButton(messages("button.saveAndContinue"))
+    }
+
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/change_contact_name.scala.html:86]: Done.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/check_contact_details.scala.html [hunk 8]
```diff
@@ -56,7 +56,7 @@
         @cyaRow(
             messages(s"$key.cyaLabel.contactName"),
             Html(contactDetails.contactName.value),
-            Some(cyaChange(messages(s"$key.cyaChange.contactName"), "#")),
+            Some(cyaChange(messages(s"$key.cyaChange.contactName"), routes.RepresenteeController.changeContactName().url)),
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/check_contact_details.scala.html:59]: Aha ok.

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**